lib/pbench/server/auth/auth.py

```diff
@@ -11,170 +12,168 @@
-            auth_token: Token to authenticate
-            oidc_client: OIDC client to call to authenticate the token
-        Returns:
-            True if the verification succeeds else False
-        """
-        identity_provider_pubkey = oidc_client.get_oidc_public_key(auth_token)
-        try:
-            oidc_client.token_introspect_offline(
-                token=auth_token,
-                key=identity_provider_pubkey,
-                audience=oidc_client.client_id,
-                options={
-                    "verify_signature": True,
-                    "verify_aud": True,
-                    "verify_exp": True,
-                },
-            )
-            return True
-        except (
-            jwt.ExpiredSignatureError,
-            jwt.InvalidTokenError,
-            jwt.InvalidAudienceError,
-        ):
-            Auth.logger.error("OIDC token verification failed")
-            return False
-        except Exception:
-            Auth.logger.exception(
-                "Unexpected exception occurred while verifying the auth token {}",
-                auth_token,
-            )
-
-        if not oidc_client.TOKENINFO_ENDPOINT:
-            Auth.logger.warning("Can not perform OIDC online token verification")
-            return False
-
-        try:
-            token_payload = oidc_client.token_introspect_online(
-                token=auth_token, token_info_uri=oidc_client.TOKENINFO_ENDPOINT
-            )
-        except OpenIDClientError:
-            return False
-
-        return "aud" in token_payload and oidc_client.client_id in token_payload["aud"]
+    except jwt.InvalidTokenError:
+        pass  # Ignore this silently; client is unauthenticated
+    except Exception as e:
+        logger.exception(
+            "Unexpected exception occurred while verifying the auth token {!r}: {}",
+            auth_token,
+            e,
+        )
+    return None
+
+
+def verify_third_party_token(auth_token: str, oidc_client: OpenIDClient) -> bool:
+    """
+    Verify a token provided to the Pbench server which was obtained from a
+    third party identity provider.
+    Args:
+        auth_token: Token to authenticate
+        oidc_client: OIDC client to call to authenticate the token
+    Returns:
+        True if the verification succeeds else False
+    """
+    identity_provider_pubkey = oidc_client.get_oidc_public_key(auth_token)
+    try:
+        oidc_client.token_introspect_offline(
+            token=auth_token,
+            key=identity_provider_pubkey,
+            audience=oidc_client.client_id,
+            options={
+                "verify_signature": True,
+                "verify_aud": True,
+                "verify_exp": True,
+            },
+        )
+        return True
+    except (
+        jwt.ExpiredSignatureError,
+        jwt.InvalidTokenError,
+        jwt.InvalidAudienceError,
+    ):
+        logger.error("OIDC token verification failed")
+        return False
+    except Exception:
```

Comment:
FWIW, if PR https://github.com/distributed-system-analysis/pbench/pull/3091 lands first, we'll need to leverage that work here. Otherwise, that PR will need to fix this location.